src/Integration.Vsix.UnitTests/SonarLintDaemon/DaemonIssueAdapterTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarLint.VisualStudio.Integration.Vsix;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.SonarLintDaemon
+{
+    [TestClass]
+    public class DaemonIssueAdapterTests
+    {
+        [TestMethod]
+        public void Ctor_NullIssue_Throws()
+        {
+            Action act = () => new DaemonIssueAdapter(null, "line text", "hash");
+
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("sonarLintIssue");
+        }
+
+        [TestMethod]
+        public void Ctor_NullPathAndHash_NoException()
```

Comment:
Nitpicking: I think the behavior stated in the test's name is that one that should be asserted. So if it says "NoException" - then I'd expect to see act.Should().NotThrow();
Otherwise, I think it should be renamed to reflect the actual assertion, e.g. Ctor_NullPathAndHash_PathAndHashAreSetToNull, something like that.